content/python/concepts/functions/functions.md

```diff
@@ -84,142 +49,14 @@ returned_value = check_leap_year(year_to_check)
-
-```py
-def north_american_capitals(**kwargs):
-  for country in kwargs:
-    print(country + ": " + kwargs[country])
-
-north_american_capitals(canada="Ottawa", us="Washington D.C.", mexico="Mexico City")
-```
-
-The output of the function call will be:
-
-```shell
-canada: Ottawa
-us: Washington, D.C.
-mexico: Mexico City
-```
-
-When defining a function, both forms of argument packing can be used. However, `args` must always precede `kwargs`.
-
-## Dynamically Passing Arguments
-
-When many arguments need to be passed into a function, it can be tedious to type them out individually. Instead, "argument unpacking" can be used to pass positional or keyword arguments dynamically.
-
-```codebyte/py
-def print_biography(name, country, age, hobby):
-  print("Hello! My name is " + name + " and I'm " + str(age) +
-        " years old. I come from " + country + ", and in my
-        free time I like to " + hobby + ".")
-
-adam = {
-  "name": "Adam",
-  "country": "Germany",
-  "age": 23,
-  "hobby": "paint"
-}
-
-print_biography(**adam)
-```
-
-## Recursion
-
-Functions in Python can call themselves — a concept known as "recursion". Recursion provides an elegant way to solve problems by breaking them down into smaller, more easily-solvable parts, and using those parts to build up a solution.
-
-Consider the Fibonacci sequence, whose first two terms are explicitly defined to be 0 and 1. Each subsequent term is constructed by taking the sum of the previous two terms. Thus, the first six terms of the sequence are 0, 1, 1, 2, 3, and 5.
-
-Defining a function that prints the `n`-th Fibonacci number is most easily achieved using recursion.
-
-```codebyte/py
-def fibonacci(n):
-  if n <= 1:
-    # Base cases
-    return n
-  else:
-    # Sum of the previous two terms
-    return fibonacci(n - 2) + fibonacci(n - 1)
-
-for i in range(6):
-  print(fibonacci(i))
-```
-
-Inside the `else` block of the function definition, two recursive calls to `fibonacci()` (representing the two previous numbers) are added and returned. Once `n` is equal to `0` or `1`, the base case (the `if` block) runs instead.
-
 ## Higher-Order Functions
 
 In Python, functions are treated as first-class objects. This means that they can be assigned to variables, stored in data structures, and passed to or returned from other functions.
 
-A function is considered to be of "higher-order" if it takes a function as a parameter or returns a function. One example is the built-in `filter()` function:
+Functions are considered to be "higher-order" values because they can be used as parameters or returns values for other functions. One example is the built-in `filter()` function:
```

Comment:
```suggestion
Functions are considered to be "higher-order" values because they can be used as parameters or return values for other functions. One example is the built-in `filter()` function:
```